File: HeartForCharity/HeartForCharity.Services/Database/Category.cs
using HeartForCharity.Model.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HeartForCharity.Services.Database
{
    public class Category
    {
        [Key]
        public int CategoryId { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = null!;

        [MaxLength(500)]
        public string? Description { get; set; }

        public CategoryAppliesTo AppliesTo { get; set; } = CategoryAppliesTo.Both;

        public virtual ICollection<Campaign> Campaigns { get; set; } = new List<Campaign>();
        public virtual ICollection<VolunteerJob> VolunteerJobs { get; set; } = new List<VolunteerJob>();
    }
    
}

File: HeartForCharity/UI/heartforcharity_desktop/lib/model/responses/category.dart
class Category {
  final int categoryId;
  final String name;
  final String? description;
  final String? appliesTo;

  Category({
    this.categoryId = 0,
    this.name = '',
    this.description,
    this.appliesTo,
  });

  factory Category.fromJson(Map<String, dynamic> json) => Category(
        categoryId: json['categoryId'] ?? 0,
        name: json['name'] ?? '',
        description: json['description'],
        appliesTo: json['appliesTo'],
      );

  Map<String, dynamic> toJson() => {
        'categoryId': categoryId,
        'name': name,
        'description': description,
        'appliesTo': appliesTo,
      };
}
